### ISSUE TITLE
Open numpad near touches in number puzzles?

### ISSUE THREAD

**Issue by omgitsraven** [chrisboyle/sgtpuzzles#496]
Could there be a mode where, when selecting a cell in a number puzzle, rather than moving your hand all the way to the row of numbers at the bottom of the screen, an interface for choosing a number could temporarily open right near where you just tapped?

I figure this could either take the form of a 'numpad' (like for dialing a phone), or maybe even a radial menu that opens around your finger, so that you can select a number by sliding your touch from that cell into the desired number before releasing your finger.

(I know this is a rather bigger change than some other suggestions, but I thought I'd at least bring it up anyway!)

**Comment by JohnGlassmyer** [comment]
i have myself also thought a radial thing around the touch (perhaps around a long-press?) would be natural. how would you place it, though, when you are pressing near the edge of the screen or board?

**Comment by omgitsraven** [comment]
Yeah, that's always the tricky part.  It could either just spring out off-center from your thumb (not ideal but not a dealbreaker imo), or maybe it actually opens like a 180-degree 'fan', in whatever direction there's room for?  (might be hard to fit 0-9 characters into just a half circle, though.)

**Comment by JohnGlassmyer** [comment]
or maybe we could say "OK Google, 9." to place a 9.

**Comment by omgitsraven** [comment]
Not a bad option too, but I'd still like something that I could do on the bus...

**Comment by JohnGlassmyer** [comment]
does this apply to Filling? how would selecting numbers work with regards to dragging to select cells? as for mechanisms to trigger the popup or whatever, i wonder if long-press on an existing selection would make sense. the other thing i think of is double-tapping a cell but that wouldn't work for Filling. can you think of other mechanisms to trigger the thing?

**Comment by JohnGlassmyer** [comment]
maybe long-press on a selection for Filling, and long-press on any cell for games like Solo that don't have multi-cell selection..?

**Comment by omgitsraven** [comment]
for Filling in particular, I think the dragging-to-select could just automatically paste the number of any already-filled cell that gets included in the selection (as in, you'd begin your selection on an already-filled cell, and it would bring that number with you)

**Comment by JohnGlassmyer** [comment]
i have thoughts in response to your last comment re spreading in a dragged selection in Filling, but i'll put them instead in the other issue (#500) you opened on that

**Comment by JohnGlassmyer** [comment]
also it would be unfortunate if this finger-proximate number-picking feature were to work only in some games, and not in others, involving filling cells with numbers.
